docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,192 @@
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account-summary <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-subaccount <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - autogenerate-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-seed-prase "twelve words goes here" network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-account <account-id> beneficiary <beneficiary-account-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - list-keys <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - add-key <account-id>
+    - autogenerate-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-seed-prase "twelve words goes here" network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-key <account-id> <public-key> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - TODO: "import account" (aka "login") commands
+```
+
+```
+contract
+  - call-function
+    - as-read-only <account-id> <function-name> <function-args> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+    - as-transaction <account-id> <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount> network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+  
+  - deploy <account-id> use-file <path-to-wasm-file>
+
+    - with-init-call <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount>
+      - transaction signature options here (see below)
+    - without-init-call
+      - transaction signature options here (see below)
+
+  - download-wasm <account-id> to-folder <path-to-download-folder> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - inspect-storage <account-id> key-prefix <storage-key-prefix> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+```
+
+```
+tokens <owner-account-id>
+  - send-near <receiver-account-id> <amount-in-NEAR> network <"mainnet"|"testnet"|...>
```

Comment:
Should there be a `send-yocto` as well?  With `near-units` installed you can do `send-yocto <A> "$(near-units 0.05 N)"`

Reply:
@willemneal near-cli-rs already supports yN/yNEAR/yoctoNEAR/yoctoN (case-insensitive):

https://github.com/near/near-cli-rs/blob/226bf02bdbc46e2a03c857879d6c2d632e2b27de/src/common.rs#L246-L248

Thus, you can already do `send-near frol.near 1yoctoNEAR`. There is no need to have a separate command for that.

Reply:
Ah well `<amount-in-NEAR>` made it seem that it only took units of near.  Glad to see there is support.  Though it seems you only support N and yN, but no `mN`.

Reply:
We could add more prefixes in the future when there is evidence that it is necessary. I have not seen use-cases when someone wants to attach or send mN amounts.

Reply:
It comes in handy when you need to provide a small amount to cover storage.  But I agree it's a rare instance.

Reply:
BTW, if you will ever feel limited by the supported prefixes and want to use `near-units`, you can still use

```
send-near <A> "$(near-units 5 mN) yoctoNEAR"
```